ShareX.UploadersLib/FileUploaders/Uplea.cs

```diff
@@ -0,0 +1,231 @@
+        {
+            return new Uplea() { UpleaApiKey = config.UpleaApiKey };
+        }
+
+        public override TabPage GetUploadersConfigTabPage(UploadersConfigForm form) => form.tpUplea;
+    }
+
+    [Localizable(false)]
+    public sealed class Uplea : FileUploader
+    {
+        private const string upleaBaseUrl = "http://api.uplea.com/api/";
+        public string UpleaApiKey { get; set; }
+
+        public UpleaGetUserInformationResponse GetUserInformation(string apiKey)
+        {
+            var upleaGetUserInformationResponse = JsonConvert.DeserializeObject<UpleaGetUserInformationResponse>(GetUpleaResponse<UpleaGetUserInformationRequest>(new UpleaGetUserInformationRequest() { ApiKey = apiKey }));
+
+            if (upleaGetUserInformationResponse.Status)
+            {
+                return upleaGetUserInformationResponse;
+            }
+
+            return new UpleaGetUserInformationResponse();
+        }
+
+        private UpleaNode GetBestNode()
+        {
+            var getBestNodeResponse = JsonConvert.DeserializeObject<UpleaGetBestNodeResponse>(SendRequest(HttpMethod.POST, upleaBaseUrl + "get-best-node"));
+            return new UpleaNode(getBestNodeResponse.Result.Name, getBestNodeResponse.Result.Token);
+        }
+
+        public string GetApiKey(string username, string password)
+        {
+            var upleaGetApiKeyResponseStr = GetUpleaResponse(new UpleaGetApiKeyRequest() { Username = username, Password = password });
+
+            if (!string.IsNullOrEmpty(upleaGetApiKeyResponseStr))
+            {
+                try
+                {
+                    var upleaGetApiKeyResponse = JsonConvert.DeserializeObject<UpleaGetApiKeyResponse>(upleaGetApiKeyResponseStr);
+
+                    if (upleaGetApiKeyResponse.Status)
+                    {
+                        return upleaGetApiKeyResponse.Result.ApiKey;
+                    }
+                }
+                catch (JsonSerializationException ex)
+                {
+                    // For some reason the Uplea API is supposed to return a single object in the result property of the response, but when 
+                    // there is an error it returns an empty array. This is causing deserialziation to fail. Do we want to just query the JSON 
+                    // status before trying to deserialize? 
+
+                    System.Diagnostics.Debug.WriteLine("Deserialization of UpleaGetApiKeyResponse failed: {0}", ex.Message);
+
+                }
+            }
+
+            return string.Empty;
+        }
+
+        private string GetUpleaResponse<T>(T upleaRequest) where T : IUpleaRequest
+        {
+            Dictionary<string, string> parameters = new Dictionary<string, string>();
+            parameters.Add("json", JsonConvert.SerializeObject(upleaRequest));
+            return SendRequestURLEncoded(upleaBaseUrl + upleaRequest.RequestType, parameters);
```

Comment:
Yes, unfortunately SendRequestJSON didn't work for this. This API is very picky. I couldn't get it to respond successfully when using application/json as the Content-Type. GetUpleaResponse and consequently SendRequestURLEncoded is only being used for the non-Upload requests to Uplea, which even according to their API documentation is different than the Upload request. I admit the language on their API page suggests JSON encoding.